### ISSUE TITLE
HLTV

### ISSUE THREAD

**Issue by uGamers** [GameServerManagers/LinuxGSM#1294]
Can add a script for HLTV? for cs 1.6 cs source csgo and any other servers that can use it?

**Comment by UltimateByte** [comment]
Can you be more specific? We're not on twitter, this is a development place, so if you got a project or a request, you need to provide details.

**Comment by uGamers** [comment]
I will keep looking for info on it. 

Check this out http://www.slipgate.de/download/HLTV-Readme.txt

**Comment by UltimateByte** [comment]
So this is for goldsource engine? Like cs 1.6 etc, that don't offer sourcetv?

**Comment by uGamers** [comment]
Yes GoldSource. It runs along side the HLDS Server providing a way to have spectators without using slots on the server besides the 1 the HLTV uses to connect to server.

**Comment by uGamers** [comment]
Files needed
1. hltv.cfg
`// HLTV Proxy configuration file


// HLTV proxy runs this file on start up
// This file should only be edited if you want to broadcast a game

// set HLTV proxy name as shown in score board
name "Name the HLTV on Player list"

// set HLTV name, how it should appear in game server browsers
hostname                "Server Browser Name"

// set offline info text clients will see as reject reason if HLTV isn't
//broadcasting yet
offlinetext "Sorry, game is delayed. Please try again later."

// delays broadcasting for 30 seconds
delay 30.0

// allow 3.5 KByte/sec as client rate. This is good a value
// for internet broadcasts. On LAN you may set this value to 10000
maxrate 3500

// log HLTV console in proxy.log
// logfile 1

// local chatting for HLTV spectators enabled
chatmode 1

// if game server is password protected, enable this line
serverpassword  ""

// proxy's adim password for rcon, commentator etc.
adminpassword   ""

// show message for 5 seconds each 12 seconds in center of X axis (-1) and
// above help text bar (0.85). Color given as hexadecimal RGBA .
//loopcmd 1 120 localmsg "Voce esta assintindo a LiBra-TV -
//www.ligabrasileira.com.br" 5 -1 0.85 FFA000FF

// hltv.tga will be shown instead of the default HLTV logo in spectator GUI
// bannerfile "hltv.tga"

// these commands will be executed on connecting spectator client and may be
//used
// to adjust settings for HLTV (for example voice parameters)
signoncommands "voice_scale 2; voice_overdrive 16; volume 0.5; //echo Voice
//adjusted for HLTV"

maxclients "35"

connect server|ip <This is the server you want HLTV to connect to>

echo hltv.cfg loaded.
`
2. I use this to start but you would prob do something like this within csserver script.
`tmux kill-session -t hltv
tmux new-session -d -s hltv ./hltv.lib -console -ip 000.000.000.000 -port 27020
echo "Server Started Successfully" `

IP == What you want the HLTV IP to be in Server Browser. 

3. I use this to stop
`tmux kill-session -t hltv
echo "Server Stopped"`

Anyways Just thought some might need or want it in this script 💃

**Comment by borzaka** [comment]
For CS:GO, there is an [ESL serverconfig](https://play.eslgaming.com/download/26251762/), which contains an `eslgotv.cfg`:

```
// ESL - www.eslgaming.com
// GOTV Config
// 14.01.2016


sv_hibernate_postgame_delay 180

tv_allow_camera_man 1        	// Auto director allows spectators to become camera man
tv_allow_static_shots 1        	// Auto director uses fixed level cameras for shots
tv_autorecord 1            	// Automatically records all games as GOTV demos.
tv_chatgroupsize 0        	// Set the default chat group size
tv_chattimelimit 8        	// Limits spectators to chat only every n seconds
tv_debug 0                	// GOTV debug info.
tv_delay 90                	// GOTV broadcast delay in seconds
tv_delaymapchange 1        	// Delays map change until broadcast is complete
tv_deltacache 2            	// Enable delta entity bit stream cache
tv_dispatchmode 1            	// Dispatch clients to relay proxies: 0=never, 1=if appropriate, 2=always
tv_enable 1                	// Activates GOTV on server.
tv_maxclients 10            	// Maximum client number on GOTV server.
tv_maxrate 20000            	// Max GOTV spectator bandwidth rate allowed, 0 == unlimited
tv_name "ESL - GOTV"    	// GOTV host name
tv_overridemaster 0        	// Overrides the GOTV master root address.
tv_port 27020            	// Host SourceTV port
tv_relaypassword schmetterling 	// GOTV password for relay proxies
tv_relayvoice 0            	// Relay voice data: 0=off, 1=on
tv_snapshotrate 24        	// Snapshots broadcasted per second
tv_timeout 60            	// GOTV connection timeout in seconds.
tv_title "ESL - GOTV"    	// Set title for GOTV spectator UI
tv_transmitall 1            	// Transmit all entities (not only director view)

say "> ESL GOTV Config loaded - 14.01.2016 <"
```

**Comment by MacGyver27** [comment]
Can we please have this problem addressed? Or please point me where exactly I should do the script execution. 

I would like to bind HLTV startup with the server itself - so the HLTV will reconnect on every start/restart etc.

Thanks!

**Comment by MacGyver27** [comment]
Any progress on this?
